### PR TITLE
Khanayan123/enable nodejs consistency test

### PR DESCRIPTION
**Pull request by khanayan123**
## Motivation

Enable config consistency tests for node.js

## Changes

Enabled config consistency tests for node.js along with some changes to make tests follow spec

### CODE REVIEW

File: manifests/nodejs.yml [hunk 1]
```diff
@@ -557,11 +558,11 @@ tests/:
       TestAdmisionControllerProfiling: *ref_5_22_0
   parametric/:
     test_config_consistency.py:
-      Test_Config_RateLimit: missing_feature
-      Test_Config_TraceAgentURL: *ref_5_22_0
-      Test_Config_TraceEnabled: missing_feature
+      Test_Config_RateLimit: *ref_5_25_0
+      Test_Config_TraceAgentURL: *ref_5_25_0
+      Test_Config_TraceEnabled: *ref_3_2_0
```
Comment (simon-id) [inline manifests/nodejs.yml:563]: did you try running the tests for all those "old" versions ?
I can see `3.2.0`, `3.13.0`, and `3.0.0` being added.
(ask me if you need help on how to do that)

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
There are failures in the CI related to changes made in this PR. I set it back to draft, you can set it to normal once failure are fixed.

File: manifests/nodejs.yml (hunk 1 above)
Comment (khanayan123) [reply manifests/nodejs.yml:563]: Yes, the tests are currently passing in the CI. I looked into the earliest implementation of those tests to verify their success. The remaining tests required some modifications and are pending the next release, once the PR implementing those changes in dd-trace-js get released.

Comment (khanayan123) [reply manifests/nodejs.yml:563]: Unless you mean manually running them against those versions locally?

Comment (simon-id) [reply manifests/nodejs.yml:563]: Yeah no I meant running those tests (locally or not) with old versions of the library to make sure the version you're putting here is correct.
The system tests CI by default will only test on the latest version released and on the master branch. It won't test old versions for you.

>  The remaining tests required some modifications and are pending the next release, & so are scheduled to be run once the PR implementing those changes in dd-trace-js get released.

If your PR is in master, it should already be tested in this CI, see the `Testing the test / System Tests (nodejs, dev)` jobs. "dev" -> master, "prod" -> latest release.

Comment (khanayan123) [reply manifests/nodejs.yml]: Ok updated manifest file after testing those versions locally

File: utils/_context/_scenarios/__init__.py [hunk 2]
```diff
@@ -441,6 +441,7 @@ def all_endtoend_scenarios(test_object):
             "DD_TRACE_HTTP_CLIENT_ERROR_STATUSES": "200-201,202",
             "DD_SERVICE": "service_test",
             "DD_TRACE_KAFKA_ENABLED": "false",  # Using Kafka as is the most common endpoint and integration(missing for PHP).
+            "DD_TRACE_KAFKAJS_ENABLED": "false",  # In Node the integration is kafkajs.
```
Comment (cbeauchesne) [inline utils/_context/_scenarios/__init__.py:444]: Why is it different in node ? this name isn't specified in a RFC ?

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
There are still failure in CI on the activated tests.

File: utils/_context/_scenarios/__init__.py (hunk 2 above)
Comment (khanayan123) [reply utils/_context/_scenarios/__init__.py:444]: I don't know if there was a specific naming requirement in an RFC when this integration was implemented but node just refers to the integration internally as the npm package name:

https://www.npmjs.com/package/kafkajs

File: tests/parametric/test_telemetry.py [hunk 3]
```diff
@@ -148,23 +148,24 @@ def test_library_settings(self, library_env, test_agent, test_library):
...
-        assert configuration_by_name.get("DD_TRACE_RATE_LIMIT").get("value") == 10
+        assert configuration_by_name.get("DD_TRACE_RATE_LIMIT").get("value") == "10"
         assert (
             configuration_by_name.get("DD_TRACE_HEADER_TAGS").get("value") == "User-Agent:my-user-agent,Content-Type."
         )
-        assert configuration_by_name.get("DD_TRACE_ENABLED").get("value") == False
+        assert configuration_by_name.get("DD_TRACE_ENABLED").get("value") == True
         assert configuration_by_name.get("DD_TRACE_OBFUSCATION_QUERY_STRING_REGEXP").get("value") == "\d{3}-\d{2}-\d{4}"
-        assert configuration_by_name.get("DD_TRACE_LOG_DIRECTORY").get("value") == "/some/temporary/directory"
+        if context.library != "nodejs":
```
Comment (cbeauchesne) [inline tests/parametric/test_telemetry.py:161]: Could you add a comment explaining why is it different in node ?

File: tests/test_config_consistency.py [hunk 4]
```diff
@@ -327,7 +328,7 @@ class Test_Config_IntegrationEnabled_False:
     """ Verify behavior of integrations automatic spans """
 
     def setup_kafka_integration_enabled_false(self):
-        self.r = weblog.get("/kafka/produce")
+        self.r = weblog.get("/kafka/produce", params={"topic": "random"}, timeout=60)
```
Comment (cbeauchesne) [inline tests/test_config_consistency.py:331]: Why is there such a long timeout ?

Comment (khanayan123) [reply tests/test_config_consistency.py]: Based it off another test that had timeout as 60

File: tests/parametric/test_telemetry.py (hunk 3 above)
Comment (cbeauchesne) [reply tests/parametric/test_telemetry.py:165]: > these settings are not supported by the nodejs sdk

Could you split the test method in two separated test method, and decorate the one with the unsupported part with `@missing_feature` ?

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

File: tests/test_config_consistency.py [hunk 5]
```diff
@@ -327,7 +328,7 @@ class Test_Config_IntegrationEnabled_False:
     """ Verify behavior of integrations automatic spans """
 
     def setup_kafka_integration_enabled_false(self):
-        self.r = weblog.get("/kafka/produce", params={"topic": "Something"})
+        self.r = weblog.get("/kafka/produce", params={"topic": "Something"}, timeout=1)
```
[the remark this answers is not in the document]
Comment (cbeauchesne) [reply tests/test_config_consistency.py:331]: confirmed : https://github.com/DataDog/system-tests/actions/runs/11690764155/job/32557083458?pr=3144#step:24:57

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
Per comment, could you remove the timeout as it cuase fails on other langs.

Updating the brnach will also solve some CI errors.

File: tests/test_config_consistency.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (khanayan123) [reply tests/test_config_consistency.py]: removed

**Review by cbeauchesne — APPROVED** [review]

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
There are failure related to the change.

**Comment by cbeauchesne** [comment]
I set back this PR to draft, to unstress my notifications stack 😄 . You can set it back to normal once it;s ready for review. If you need any help, please reach me on slack.

**Comment by khanayan123** [comment]
> I set back this PR to draft, to unstress my notifications stack 😄 . You can set it back to normal once it;s ready for review. If you need any help, please reach me on slack.

I think it was just test flakiness, after rerunning those tests are passing now

**Pull request merged**